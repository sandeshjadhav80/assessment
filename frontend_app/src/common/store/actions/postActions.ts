export const ADD_POST = 'ADD_POST';
export const FILTER_POST = 'FILTER_POST';