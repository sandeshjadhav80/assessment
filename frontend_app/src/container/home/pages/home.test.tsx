import React, { Component }  from 'react';
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import configureStore from 'redux-mock-store' //ES6 modules
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from './home';
import {ADD_POST} from './../../../common/store/actions/postActions';

const middlewares: any = []
const mockStore = configureStore(middlewares);
let container:any = null;
let store: any;
let component:any;
beforeEach(() => {
  
    store = mockStore({
        age: 22,
        postList: [],
        filterPostList: [],
    });

    store.dispatch = jest.fn();

    component = renderer.create(
        <Provider store={store}>
            <Home postList={[]} filterPostList={[]} addPost={''} dispatch={''} />
        </Provider>
    );
});


 it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 

 