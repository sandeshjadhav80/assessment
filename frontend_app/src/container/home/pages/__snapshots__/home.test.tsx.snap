// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with given state from Redux store 1`] = `
<div>
  <div>
    <h1>
      Header
    </h1>
  </div>
  <div
    className="content"
  >
    <div
      className="post-wapper"
    >
      <div
        className="search-box-wrapper clearfix"
      >
        <div
          className="search-box"
        >
          <div
            className="search-icon"
          >
            <img
              alt="Logo"
              height="30px"
              onClick={[Function]}
              src="https://lh3.googleusercontent.com/proxy/WK4x-4FU5oYalQ4cV0nTD8RUGTMA9blAhM9X150CY4QCjR4O8ZhFJAIZV5npEoRGfOfvKcWdiib2IZJqOuete0AMEzQOkGA_lVqEgSrb8Az4wz-9PY4"
              style={
                Object {
                  "cursor": "pointer",
                  "marginTop": "5px",
                }
              }
              width="30px"
            />
          </div>
          <div
            className="input-text"
          >
            <input
              className="search-text-box"
              onChange={[Function]}
              placeholder="Search..."
              type="text"
              value=""
            />
          </div>
          <div
            className="clear-icon"
          >
            <img
              alt="Logo"
              height="30px"
              onClick={[Function]}
              src="https://cdn.iconscout.com/icon/premium/png-256-thumb/clear-16-216586.png"
              style={
                Object {
                  "cursor": "pointer",
                  "marginTop": "5px",
                }
              }
              width="30px"
            />
          </div>
        </div>
      </div>
      <div
        className="post-content clearfix"
      >
        <div
          className="post-tab-wrapper clearfix"
        >
          <button
            className="tab-buttons"
            onClick={[Function]}
          >
            New Post
          </button>
          <button
            className="tab-buttons"
            onClick={[Function]}
          >
            All Published Post
          </button>
        </div>
        <div
          className="post-form-wrapper"
        >
          <div
            className="form-row"
          >
            <label>
              Title
            </label>
            <div
              className="form-row-control"
            >
              <input
                onChange={[Function]}
                style={
                  Object {
                    "borderColor": "whitesmoke",
                  }
                }
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-row"
          >
            <label>
              Text
            </label>
            <div
              className="form-row-control"
            >
              <input
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-row"
          >
            <div
              className="form-row-control"
            >
              <button
                className="btn-custom"
                onClick={[Function]}
              >
                Published
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    Footer
  </div>
</div>
`;
